app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java

```diff
@@ -356,10 +368,47 @@ public boolean onOptionsItemSelected(MenuItem item) {
             parsedData = StatisticsFragment.parsedData;
             plotGraph();
             toggleMenuItem(menu, isPlaying);
+        } else if (id == R.id.save_graph_data) {
+            saveData();
         }
         return super.onOptionsItemSelected(item);
     }
 
+    private void saveData() {
+        File importedFile = new File(importedFilePath);
+        FilePathUtil.setupPath();
+        String fileName = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS", Locale.getDefault()).format(SystemClock.currentThreadTimeMillis());
+        File dst = new File(Environment.getExternalStorageDirectory().getAbsolutePath() +
+                File.separator + CSV_DIRECTORY + File.separator + fileName + ".csv");
+        if (!dst.exists()) {
+            try {
+                transfer(importedFile, dst);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void transfer(File src, File dst) throws IOException {
```

Comment:
What is this doing?

Reply:
It actually makes a copy of the data being used currently with the date-time as name of the file in the created NeuroLab directory which is being accessed by the DataLoggerActivity. I think we can change the function name here and add some comments above. @CloudyPadmal what do you say?